src/utils/db/car_index/tests.rs

```diff
@@ -1,92 +0,0 @@
-// Copyright 2019-2023 ChainSafe Systems
-// SPDX-License-Identifier: Apache-2.0, MIT
-use super::*;
-use ahash::{AHashMap, AHashSet};
-use quickcheck_macros::quickcheck;
-
-fn query(table: &CarIndex<impl ReadAt>, key: Hash) -> Vec<FrameOffset> {
-    table.lookup_hash(key).unwrap().into_vec()
-}
-
-fn mk_table(entries: &[(Hash, FrameOffset)]) -> CarIndex<Vec<u8>> {
-    let table_builder = CarIndexBuilder::new(entries.iter().copied());
-    let mut store = Vec::new();
-    table_builder.write(&mut store).unwrap();
-    dbg!(&store[32..32 + 8]);
-    CarIndex::open(store, 0).unwrap()
-}
-
-fn mk_map(entries: &[(Hash, FrameOffset)]) -> AHashMap<Hash, AHashSet<FrameOffset>> {
-    let mut map = AHashMap::with_capacity(entries.len());
-    for (hash, position) in entries.iter().copied() {
-        map.entry(hash)
-            .and_modify(|set: &mut AHashSet<FrameOffset>| {
-                set.insert(position);
-            })
-            .or_insert(AHashSet::from([position]));
-    }
-    map
-}
-
-#[quickcheck]
-fn lookup_singleton(key: Hash, value: FrameOffset) {
-    let table = mk_table(&[(key, value)]);
-    assert_eq!(query(&table, key), vec![value]);
-    assert_eq!(query(&table, !key), Vec::<FrameOffset>::new());
-}
-
-// Identical to HashMap<Hash, HashSet<FrameOffset>> with almost no collision
-#[quickcheck]
-fn lookup_wide(entries: Vec<(Hash, FrameOffset)>) {
-    let map = mk_map(&entries);
-    let table = mk_table(&entries);
-    for (&hash, value_set) in map.iter() {
-        assert_eq!(&AHashSet::from_iter(query(&table, hash)), value_set);
-    }
-}
-
-// Identical to HashMap<Hash, HashSet<FrameOffset>> with many collision
-#[quickcheck]
-fn lookup_narrow(mut entries: Vec<(Hash, FrameOffset)>) {
```

Comment:
I don't really understand this test

Reply:
This is your `everything_maps_to_first_10_slots` test.

Reply:
Ah, this is not `everything_maps_to_first_10_slots`.

This tests what happens when there are hash collisions. As far as I can tell, you don't have any tests for hash collisions.